devlib/native/win_native.cpp

```diff
@@ -121,6 +121,37 @@ namespace winutil {
     }
 
 
+    static auto initializeRootBuses(void) -> QMap<QString, int>
+    {
+        auto rootHubsBusesMap = QMap<QString, int>{};
+
+        auto deviceInfoSet = ::SetupDiGetClassDevs(nullptr, nullptr, nullptr, DIGCF_ALLCLASSES | DIGCF_PRESENT | DIGCF_DEVICEINTERFACE);
+        if (deviceInfoSet == INVALID_HANDLE_VALUE)
+            return {};
+
+        auto busNumber = 1;
+        DWORD index = 0;
+        SP_DEVINFO_DATA devInfoData;
+        devInfoData.cbSize = sizeof(SP_DEVINFO_DATA);
+        while (SetupDiEnumDeviceInfo(deviceInfoSet, index, &devInfoData)) {
+            index++;
+
+            SP_DEVICE_INTERFACE_DATA devInterfaceData;
+            devInterfaceData.cbSize = sizeof(SP_DEVICE_INTERFACE_DATA);
+            if (!SetupDiEnumDeviceInterfaces(deviceInfoSet, &devInfoData, &GUID_DEVINTERFACE_USB_HOST_CONTROLLER, 0, &devInterfaceData))
+                continue;
+
+            char instanceId[MAX_DEVICE_ID_LEN];
+            if (!SetupDiGetDeviceInstanceIdA(deviceInfoSet, &devInfoData, instanceId, sizeof(instanceId), nullptr)) {
+                continue;
```

Comment:
I looked at `libusb`. In the event of a `SetupDiGetDeviceInstanceIdA` failure, `libusb` just skips that `busNumber` and continue without it: [For loop](https://github.com/libusb/libusb/blob/master/libusb/os/windows_winusb.c#L1218) -> [Get Device using GUID](https://github.com/libusb/libusb/blob/master/libusb/os/windows_winusb.c#L1236) -> [Skip if !SetupDiGetDeviceInstanceIdA](https://github.com/libusb/libusb/blob/master/libusb/os/windows_winusb.c#L1256) ->[Add new bus](https://github.com/libusb/libusb/blob/master/libusb/os/windows_winusb.c#L1478). Therefore, `busNumber++` should be added there

I'm not sure that we should call `SetupDiGetDeviceInstanceIdA` one more time. I have not seen reports of this behavior anywhere. But I can add a check for the `findBusNumber` function.
```
if (rootHubsBusses.isEmpty())
    rootHubsBusses = initializeRootBuses();
```
it will try to get the buses again if an error occurs in `initializeRootBuses`. But I think this may be redundant.

Reply:
> it will try to get the buses again if an error occurs in initializeRootBuses. But I think this may be redundant.

This method will likely fail again if you call it one more time. So let's not add attempts. 

> Therefore, busNumber++ should be added there

You are going to increment `busNumber` if `SetupDiGetDeviceInstanceIdA` fails, right?

Reply:
> You are going to increment busNumber if SetupDiGetDeviceInstanceIdA fails, right?

Yep

Reply:
Changed